Neos.ContentRepository/Classes/Domain/Service/PublishingService.php

```diff
@@ -182,8 +182,11 @@ protected function doDiscardNode(NodeInterface $node, array &$alreadyDiscardedNo
         if ($possibleShadowNodeData instanceof NodeData) {
             if ($possibleShadowNodeData->getMovedTo() !== null) {
                 $parentBasePath = $node->getPath();
-                $affectedChildNodeDataInSameWorkspace = $this->nodeDataRepository->findByParentAndNodeType($parentBasePath, null, $node->getWorkspace(), null, false, true);
-                foreach ($affectedChildNodeDataInSameWorkspace as $affectedChildNodeData) {
+                $affectedChildNodeDataInWorkspaceChain = $this->nodeDataRepository->findByParentAndNodeType($parentBasePath, null, $node->getWorkspace(), null, false, true);
```

Comment:
why dont we just pass the `$node->getDimensions()` as third parameter here to filter it via sql instead of fetching all nodes (in the correct workspace but ignoring dimensions) and then filter in php via: `matchesWorkspaceAndDimensions`

Reply:
ahh i see because the workspace passed as 3rd parameter will actually be expanded into all base workspaces `collectWorkspaceAndAllBaseWorkspaces` and we only want to filter against nodes in this literal workspace.

So the php filtering in `matchesWorkspaceAndDimensions` will always be necessary - a little comment would be nice as this wasnt obvious to me ;)

Reply:
That's why I adjusted the name of the variable to make it more clear, should I still add a comment?

Reply:
Ahh i see no thats fine ^^ And i also see that we use exactly the same pattern (first query all and then filter in php) in this place:

https://github.com/neos/neos-development-collection/blob/cf4077dcdc71f96843d41f1ab350e0283fe06092/Neos.ContentRepository/Classes/Domain/Service/PublishingService.php#L185-L189